mlkem/indcpa.c

```diff
@@ -408,10 +408,10 @@ void indcpa_keypair_derand(uint8_t pk[MLKEM_INDCPA_PUBLICKEYBYTES],
   poly_getnoise_eta1_4x(skpv.vec + 0, skpv.vec + 1, e.vec + 0, e.vec + 1,
                         noiseseed, 0, 1, 2, 3);
 #elif MLKEM_K == 3
-  poly_getnoise_eta1_4x(skpv.vec + 0, skpv.vec + 1, skpv.vec + 2, e.vec + 0,
-                        noiseseed, 0, 1, 2, 3);
-  poly_getnoise_eta1_4x(e.vec + 1, e.vec + 2, pkpv.vec + 0, pkpv.vec + 1,
-                        noiseseed, 4, 5, 6, 7);
+  poly_getnoise_eta1_4x(skpv.vec + 0, skpv.vec + 1, skpv.vec + 2, pkpv.vec + 0,
+                        noiseseed, 0, 1, 2, 6);
+  poly_getnoise_eta1_4x(e.vec + 0, e.vec + 1, e.vec + 2, pkpv.vec + 1,
+                        noiseseed, 3, 4, 5, 7);
```

Comment:
This is very confusing and I'm not a fan of changing the code like this. 

Is there really no way to make this simpler? 
Intuitively, I do not understand why `IS_FRESH` doesn't work for two polynomials belonging to the same polyvec - they are still disjoint chunks of memory.

Reply:
That's unfortunately not how it works. `IS_FRESH` generates the obligation that the pointers belong to different objects.